src/models/AuthenticationUtils.ts

```diff
@@ -0,0 +1,134 @@
+            throw new Error(message);
+        }
+    }
+
+    async getTokenAPI(scope: TokenScope) 
+    {
+        let tokenScopeDecoded = scope == TokenScope.Dataplane ? this.dataPlaneTokenScope : this.armTokenScope;
+        try {
+            const cmdArguments = ["account", "get-access-token", "--resource"];
+            cmdArguments.push(tokenScopeDecoded);
+            var result: any = await this.execAz(cmdArguments);
+            let token = result.accessToken;
+            
+            // NOTE: Setting the subscription id
+            this.subscriptionId = result.subscription;
+            scope == TokenScope.ControlPlane ? this.controlPlaneToken = token : this.dataPlanetoken = token;
+            return token;
+        } 
+        catch (err:any) {
+            const message =
+                `An error occurred while getting credentials from ` + `Azure CLI: ${err.message}`;
+            throw new Error(message);
+        }
+    }
+
+    async execAz(cmdArguments: string[]): Promise<any> {
+        const azCmd = process.platform === "win32" ? "az.cmd" : "az";
+        return new Promise<any>((resolve, reject) => {
+            execFile(azCmd, [...cmdArguments, "--out", "json"], { encoding: "utf8", shell : true }, (error:any, stdout:any) => {
+                if (error) {
+                    return reject(error);
+                }
+                try {
+                    return resolve(JSON.parse(stdout));
+                } catch (err:any) {
+                    const msg =
+                        `An error occurred while parsing the output "${stdout}", of ` +
+                        `the cmd az "${cmdArguments}": ${err.message}.`;
+                    return reject(new Error(msg));
+                }
+            });
+        });
+    }
+
+    isValid(scope: TokenScope) {
+        let token = scope == TokenScope.Dataplane ? this.dataPlanetoken : this.controlPlaneToken;
+        try {
+            let header = token && jwtDecode<JwtPayload>(token);
+            const now = Math.floor(Date.now() / 1000)
+            return (header && header?.exp && header.exp + 2 > now);
+        }
+        catch(error:any) {
+            console.log("Error in getting the token");
+        }
+    }
+
+    async getDataPlaneHeader(apicallType : CallTypeForDP) : Promise<IHeaders> {
+        if(!this.isValid(TokenScope.Dataplane)) {
+            let tokenRes:any = await this.getTokenAPI(TokenScope.Dataplane);
+            this.dataPlanetoken = tokenRes;
+        }
+        let headers: IHeaders = {
+            'content-type': ContentTypeMap[apicallType] ?? 'application/json',
+            'Authorization': 'Bearer '+ this.dataPlanetoken
+        };
+        return headers;
+    }
+
+    async armTokenHeader() {
+        // right now only get calls from the ADO, so no need of content type for now for the get calls.
+        var tokenRes:any = await this.getTokenAPI(TokenScope.ControlPlane);
```

Comment:
lets add isValid here because we call this in the authorise too and get the controlPlane token while initialisation itself.

Reply:
Will take this later